cmd/dagger/module.go

```diff
@@ -59,6 +59,31 @@ const (
 	defaultModuleSourceDirName = "."
 )
 
+// if the source root path already has some files
+// then use `srcRootPath/.dagger` for source
+func inferSourcePathDir(srcRootPath string) (string, error) {
+	list, err := os.ReadDir(srcRootPath)
+	if err != nil {
+		return "", err
+	}
+
+	for _, l := range list {
+		// .dagger already exist, return that
+		if l.Name() == ".dagger" {
+			return ".dagger", nil
+		}
+
+		// ignore hidden files
+		if strings.HasPrefix(l.Name(), ".") {
+			continue
+		}
+
+		return ".dagger", nil
+	}
+
+	return "", nil
```

Comment:
```suggestion
	return ".", nil
```

Right? The extra handling from having to handle the empty string is a bit confusing.

Why can't we just ignore `dagger.json` in this loop to get the desired behavior?

Reply:
I restructured the code a bit more to make this work. While doing so I also found and fixed one more scenario that involved initializing in a sub-dir.

I have also added more testcases to cover the sub-dir scenario and I hope this makes sense now.

thank you for your review comments.